Modules/@babylonjs/react-native-iosandroid/android/src/main/java/com/babylonreactnative/EngineView.java

```diff
@@ -67,34 +71,50 @@ public void surfaceDestroyed(SurfaceHolder holder) {
     public void setAntiAliasing(Integer value) {
         BabylonNativeInterop.updateMSAA(value);
     }
+
+    public void setIsTopMost(Boolean isTopMost) {
+        setIsTransparentAndIsTopMost(this.isTransparent, isTopMost);
+    }
     // ------------------------------------
     // TextureView related
-
     public void setIsTransparent(Boolean isTransparent) {
-        if (isTransparent) {
-            if (this.opaqueSurfaceView != null) {
-                this.opaqueSurfaceView.setVisibility(View.GONE);
-                this.opaqueSurfaceView = null;
-            }
-            if (this.transparentTextureView == null) {
-                this.transparentTextureView = new TextureView(this.getContext());
-                this.transparentTextureView.setLayoutParams(EngineView.childViewLayoutParams);
-                this.transparentTextureView.setSurfaceTextureListener(this);
-                this.transparentTextureView.setOpaque(false);
-                this.addView(this.transparentTextureView);
-            }
+        setIsTransparentAndIsTopMost(isTransparent, this.isTopMost);
+    }
+
+    private void setIsTransparentAndIsTopMost(Boolean isTransparent, Boolean isTopMost) {
+        if (this.surfaceView != null) {
+            this.surfaceView.setVisibility(View.GONE);
+            this.surfaceView = null;
+        }
+        if (this.transparentTextureView != null) {
+            this.transparentTextureView.setVisibility(View.GONE);
+            this.transparentTextureView = null;
+        }
+        if (isTransparent && !isTopMost) {
+            this.transparentTextureView = new TextureView(this.getContext());
+            this.transparentTextureView.setLayoutParams(EngineView.childViewLayoutParams);
+            this.transparentTextureView.setSurfaceTextureListener(this);
+            this.transparentTextureView.setOpaque(false);
+            this.addView(this.transparentTextureView);
         } else {
-            if (this.transparentTextureView != null) {
-                this.transparentTextureView.setVisibility(View.GONE);
-                this.transparentTextureView = null;
+            this.surfaceView = new SurfaceView(this.getContext());
+            this.surfaceView.setLayoutParams(EngineView.childViewLayoutParams);
+            SurfaceHolder surfaceHolder = this.surfaceView.getHolder();
+            if (isTransparent) {
+                surfaceHolder.setFormat(PixelFormat.TRANSPARENT);
             }
-            if (this.opaqueSurfaceView == null) {
-                this.opaqueSurfaceView = new SurfaceView(this.getContext());
-                this.opaqueSurfaceView.setLayoutParams(EngineView.childViewLayoutParams);
-                this.opaqueSurfaceView.getHolder().addCallback(this);
-                this.addView(this.opaqueSurfaceView);
+            if (isTopMost || isTransparent) {
```

Comment:
This can just be `if (isTopMost)` right? Otherwise we are saying we want to catch the case where `isTopMost` is false and `isTransparent` is true, but that's the first condition of the higher level logic that would use `TextureView`.

Reply:
Fixed.